cmd/guacgql/cmd/server.go

```diff
@@ -51,6 +53,20 @@ const (
 	neptuneServiceName = "neptune-db"
 )
 
+type backendFunc func(context.Context) (backends.Backend, error)
+
+var getBackend map[string]backendFunc
+
+func init() {
+	if getBackend == nil {
+		getBackend = make(map[string]backendFunc)
+	}
+	getBackend[arango] = getArango
+	getBackend[neo4js] = getNeo4j
+	getBackend[inmems] = getInMem
+	getBackend[neptune] = getNeptune
+}
```

Comment:
can we maybe move this to the backends pakage that will return this map and instead do that there? I think that may be another appropriate place to put it and we can dedup the code if other CLIs or unit tests need to do this as well.

Reply:
+1 to this, if possible.

Reply:
Updated, ptal. Still need to setup `entbackend.BackendOptions` from cli flags in a 64-bit-only file.

Reply:
I was also thinking to do the registration within the package, kind of like we do for the parsers, and other pluggable things: 

https://github.com/guacsec/guac/blob/main/pkg/ingestor/parser/parser.go#L41-L51

Reply:
Yes, I've added that as you requested in the third commit on this PR:

https://github.com/guacsec/guac/pull/1226/commits/144d04501776e54e6b06d70d4f01c42ab1cef0ff#diff-1aba4420a9017b9d486c0fab38a08b9eea2e9c22e2402039d7726f26b1ccf544R164-R167
https://github.com/guacsec/guac/pull/1226/commits/144d04501776e54e6b06d70d4f01c42ab1cef0ff#diff-84928fb53202e5f8faf758692f063b2919686aec75b8d6d22381bc6e612a3e81R22-R24
And then the call uses `backend.Get` here https://github.com/guacsec/guac/pull/1226/commits/144d04501776e54e6b06d70d4f01c42ab1cef0ff#diff-c5774aa38759e9db3bd03f13cbd58be3e05839da1c2c9b66e9fa2a45f33a5661R138

The cli still needs to prepare the backend-specific options struct, and also cannot import the entbackend options struct on 32-bit (as that would pull in the whole package).

It's a bit of a false encapsulation as the options structs need to be known, I'm not sure it is better than how it was previously.

Reply:
Ah gotcha, i think that's good for now then. I do have an idea that could maybe help a bit but not blocking.